Sources/MapboxDirections/Directions.swift

```diff
@@ -152,40 +155,36 @@ open class Directions: NSObject {
                     
                     guard let disposition = try? decoder.decode(ResponseDisposition.self, from: data) else {
                         let apiError = DirectionsError(code: nil, message: nil, response: possibleResponse, underlyingError: possibleError)
-                        let response = RouteResponse(httpResponse: httpResponse, options: .route(options), credentials: self.credentials)
 
                         DispatchQueue.main.async {
-                            completionHandler(response, apiError)
+                            completionHandler(session, .failure(apiError))
                         }
                         return
                     }
                     
                     guard (disposition.code == nil && disposition.message == nil) || disposition.code == "Ok" else {
                         let apiError = DirectionsError(code: disposition.code, message: disposition.message, response: response, underlyingError: possibleError)
-                        let response = RouteResponse(httpResponse: httpResponse, options: .route(options), credentials: self.credentials)
-
                         DispatchQueue.main.async {
-                            completionHandler(response, apiError)
+                            completionHandler(session, .failure(apiError))
                         }
                         return
                     }
                     
                     let result = try decoder.decode(RouteResponse.self, from: data)
                     guard result.routes != nil else {
                         DispatchQueue.main.async {
-                            completionHandler(result, .unableToRoute)
+                            completionHandler(session, .failure(.unableToRoute))
```

Comment:
I am not sure about original reasons, but as @1ec5 mentioned in [original proposal](https://github.com/mapbox/mapbox-directions-swift/issues/414#issuecomment-586404352),  this was the case which returned an error and a waypoints at the same time. I don't know if it was done for a good reason or not, but apparently such behavior will be lost.

Reply:
as it turns out, `session` contains an options object with the waypoints.